File: Calculator/decimals/dec-binary.go
package main

import "fmt"

func main(){
	var i int
	for i = 1;i > 1;i++ {
		binary := []int{}
		var n = 20
		var a = 0.5
		var b = 0.9
		var p = n % 2
		if p >= a && p <= b{
			p = 1
		}else if p >= 1.0 && p <= 1.4{
			p = 1
			n = n/2
		}
	fmt.Println(binary)
	}
}

File: Calculator/decimals/dec-binary.py
binary = []
n = int(input("Enter the decimal number: "))

while n >= 1:
    p = n % 2
    if p >= 0.5 and p <= 0.9:
        p = 1
    elif p >= 1.0 and p <= 1.4:
        p = 1
    binary.append(int(p))
    n = n/2

while True:
    if len(binary) == 8:
        break
    else:
        binary.append(0)
        
print(binary[::-1])